pkg/controller/controllerutil/controllerutil.go

```diff
@@ -116,18 +116,26 @@ const ( // They should complete the sentence "Deployment default/foo has been ..
 
 // CreateOrUpdate creates or updates the given object obj in the Kubernetes
 // cluster. The object's desired state should be reconciled with the existing
-// state using the passed in ReconcileFn. obj must be a struct pointer so that
+// state using the passed in MutateFn. obj is just a struct pointer so that
 // obj can be updated with the content returned by the Server.
 //
+// The MutateFn is called regardless of creating or updating an object.
+//
 // It returns the executed operation and an error.
-func CreateOrUpdate(ctx context.Context, c client.Client, obj runtime.Object, f MutateFn) (OperationResult, error) {
+func CreateOrUpdate(ctx context.Context, c client.Client, obj runtime.Object, f MutateFn) (OperationResult, error) { // nolint: gocyclo
 	key, err := client.ObjectKeyFromObject(obj)
 	if err != nil {
 		return OperationResultNone, err
 	}
 
 	if err := c.Get(ctx, key, obj); err != nil {
 		if errors.IsNotFound(err) {
```

Comment:
can improve nesting and readability here a bit by inverting this:

```suggestion
if !errors.IsNotFound(err) {
    return OperationResultNone, err
}
if err := f(); err != nil {
```

Reply:
will probably fix the cyclo issue too.